src/gsi/deter_sfc_mod.f90

```diff
@@ -1407,90 +1403,55 @@ subroutine deter_sfc_gmi(dlat_earth,dlon_earth,isflg,sfcpct)
      end if
 
      klon1=int(dlon); klat1=int(dlat)
-     dx  =dlon-klon1; dy  =dlat-klat1
-     dx1 =one-dx;    dy1 =one-dy
-     w00=dx1*dy1; w10=dx1*dy; w01=dx*dy1; w11=dx*dy
 
      klat1=min(max(1,klat1),nlat_sfc); klon1=min(max(0,klon1),nlon_sfc)
      if(klon1==0) klon1=nlon_sfc
      klatp1=min(nlat_sfc,klat1+1); klonp1=klon1+1
-     if(klonp1==nlon_sfc+1) klonp1=1
-     klonp2 = klonp1+1
-     if(klonp2==nlon_sfc+1) klonp2=1
-     klon2=klon1-1
-     if(klon2==0)klon2=nlon_sfc
-     klat2=max(1,klat1-1)
-     klatp2=min(nlat_sfc,klatp1+1)
 
 !    Set surface type flag.  Begin by assuming obs over ice-free water
 
-     sfcpct = zero
-
-     jsli = isli_full(klat1 ,klon1 )
-     if(sno_full(klat1 ,klon1 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klatp1,klon1 )
-     if(sno_full(klatp1 ,klon1 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klat1 ,klonp1)
-     if(sno_full(klat1 ,klonp1 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klatp1,klonp1)
-     if(sno_full(klatp1 ,klonp1 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klatp2,klon1)
-     if(sno_full(klatp2 ,klon1 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klatp2,klonp1)
-     if(sno_full(klatp2 ,klonp1 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klatp1,klon2)
-     if(sno_full(klatp1 ,klon2 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klatp1,klonp2)
-     if(sno_full(klatp1 ,klonp2 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klat1,klon2)
-     if(sno_full(klat1 ,klon2 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klat1,klonp2)
-     if(sno_full(klat1 ,klonp2 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klat2,klon1)
-     if(sno_full(klat2 ,klon1 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     jsli = isli_full(klat2,klonp1)
-     if(sno_full(klat2 ,klonp1 ,it) > one .and. jsli == 1)jsli=3
-     sfcpct(jsli)=sfcpct(jsli)+one
-
-     sfcpct=sfcpct/12.0_r_kind
-
-!     sfcpct(3)=min(sfcpct(3),sfcpct(1))
-!     sfcpct(1)=max(zero,sfcpct(1)-sfcpct(3))
-
-     if(sfcpct(0) > 0.99_r_kind)then
-        isflg = 0
-     else if(sfcpct(1) > 0.99_r_kind)then
-        isflg = 1
-     else if(sfcpct(2) > 0.99_r_kind)then
-        isflg = 2
-     else if(sfcpct(3) > 0.99_r_kind)then
-        isflg = 3
-     else
-        isflg = 4
-     end if
 
+     grid_dist=rearth * (rlats_sfc(klatp1) - rlats_sfc(klat1))
+     n_grid=int(40 * 1000 / grid_dist) + 1
```

Comment:
I think fortran order operations will execute `(40 * 1000 / grid_dist)` as `40 * 1000` first with the product then divided by `grid_dist`.  If true, simplify this to `(40000 / grid_dist)`

Reply:
Simplified. Thanks.